core/graphql/src/main/kotlin/dev/materii/gloom/core/graphql/GraphQLDataSource.kt

```diff
@@ -0,0 +1,471 @@
+
+    // Repositories
+
+    suspend fun getRepoName(owner: String, name: String): GraphQLResponse<RepoNameQuery.Data>
+
+    suspend fun getRepoDetails(owner: String, name: String): GraphQLResponse<RepoDetailsQuery.Data>
+
+    suspend fun getRepoLicense(owner: String, name: String): GraphQLResponse<RepoLicenseQuery.Data>
+
+    suspend fun prefetchRepoTree(owner: String, name: String): GraphQLResponse<RepoTreePrefetchQuery.Data>
+
+    suspend fun getRepoIssues(
+        owner: String,
+        name: String,
+        after: String? = null,
+        states: Set<IssueState> = setOf(IssueState.OPEN)
+    ): GraphQLResponse<RepoIssuesQuery.Data>
+
+    suspend fun getRepoPullRequests(
+        owner: String,
+        name: String,
+        after: String? = null,
+        states: Set<PullRequestState> = setOf(PullRequestState.OPEN)
+    ): GraphQLResponse<RepoPullRequestsQuery.Data>
+
+    suspend fun getRepoReleases(
+        owner: String,
+        name: String,
+        after: String? = null
+    ): GraphQLResponse<RepoReleasesQuery.Data>
+
+    suspend fun getRepoForks(
+        owner: String,
+        name: String,
+        after: String? = null
+    ): GraphQLResponse<RepoForksQuery.Data>
+
+    suspend fun getRepoContributors(
+        owner: String,
+        name: String,
+        after: String? = null
+    ): GraphQLResponse<RepoContributorsQuery.Data>
+
+    suspend fun getCommits(
+        id: String,
+        branch: String? = null,
+        after: String? = null
+    ): GraphQLResponse<CommitsQuery.Data>
+
+    // Release
+
+    suspend fun getReleaseDetails(
+        owner: String,
+        name: String,
+        tag: String,
+        after: String? = null
+    ): GraphQLResponse<ReleaseDetailsQuery.Data>
+
+    // Files
+
+    suspend fun getTree(owner: String, name: String, branchAndPath: String): GraphQLResponse<TreeQuery.Data>
+
+    suspend fun getFile(owner: String, name: String, branch: String, path: String): GraphQLResponse<FileQuery.Data>
+
+    suspend fun getRawMarkdown(owner: String, name: String, branch: String, path: String): GraphQLResponse<RawMarkdownQuery.Data>
+
+}
+
+internal class NetworkGraphQLDataSource(
+    private val apolloClient: ApolloClient
+): GraphQLDataSource {
+
+
+    private fun <D: Operation.Data> ApolloCall<D>.addToken(): ApolloCall<D> {
+        return addHttpHeader("Authorization", "") // TODO: Retrieve token from storage
+    }
+
+    // Account
+
+    override suspend fun getAccountInfo(token: String): GraphQLResponse<AccountInfoQuery.Data> {
+        return apolloClient.query(AccountInfoQuery())
+            .doNotStore(true)
+            .addHttpHeader("Authorization", token)
+            .response()
+    }
+
+    override suspend fun identify(): GraphQLResponse<IdentifyQuery.Data> {
+        return apolloClient.query(IdentifyQuery())
+            .addToken()
+            .response()
```

Comment:
It might be better to make a function in this class which takes the query and returns the result. Cause right now its a lot of duplicate code. It could shorten this file a good bit
something like
```kt
return handleQuery(IdentityQuery())
```